src/main/java/ca/gc/cyber/ops/assemblyline/java/client/clients/AssemblylineClient.java

```diff
@@ -227,6 +228,15 @@ public Mono<IngestResponse> ingestBinary(BinaryFile<IngestBase> binaryIngest) {
 
     }
 
+    @Override
+    public Mono<IngestResponse> ingestAsyncBinary(AsyncBinaryFile<IngestBase> asyncBinaryIngest) {
+        return Mono.fromCallable(() -> this.multipartInserterFromAsyncBinaryIngest(asyncBinaryIngest))
+                .subscribeOn(Schedulers.boundedElastic())
+                .flatMap(multipartInserter -> post(buildUri(INGEST_URL), new ParameterizedTypeReference<>() {
+                        },
+                        multipartInserter, MediaType.MULTIPART_FORM_DATA));
+    }
+
```

Comment:
The common logic could be extract from this method and `ingestBinary()` into a method that accepts a `Supplier<BodyInserters.MultipartInserter>`.